### PR TITLE
Rename all AXLOG/printf  to AXLOGD (without .\core, .\3rdparty)

### LINKED ISSUE #2019: Rename all AXLOG to AXLOGD (there be 417 without .\core)
image

See PR: (https://github.com/axmolengine/axmol/pull/2017)

@halx99 
Can I do it or any issues for waiting?

### PR DESCRIPTION
**Pull request by aismann**
## Describe your changes
Rename all AXLOG to AXLOGD (without .\core)

## Issue ticket number and link
#2019

### For each PR
- [x] Add Copyright if it missed:   
      - `"Copyright (c) 2019-present Axmol Engine contributors (see AUTHORS.md)."`
- [x] I have performed a self-review of my code.
       
   Optional:
   - [ ] I have checked readme and add important infos to this PR.
   - [ ] I have added/adapted some tests too.
          
### For core/new feature PR
- [ ] I have checked readme and add important infos to this PR.
- [ ] I have added thorough tests.

### CODE REVIEW

**Comment by halx99** [comment]
note you can't replace only, the format style also need change from `%xx to {}`

**Comment by aismann** [comment]
> note you can't replace only, the format style also need change from `%xx to {}`

You mean this (e.g.:  `[%f, %f]` => ` [{}, {}]`)?
       `AXLOG("Invalid movement, distance = [%f, %f], disInch = %f", moveDistance.x, moveDistance.y);`
       `AXLOGD("Invalid movement, distance = [{}, {}], disInch = {}", moveDistance.x, moveDistance.y);`

**Comment by halx99** [comment]
> > note you can't replace only, the format style also need change from `%xx to {}`
> 
> You mean this (e.g.: `[%f, %f]` => ` [{}, {}]`)? `AXLOG("Invalid movement, distance = [%f, %f], disInch = %f", moveDistance.x, moveDistance.y);` `AXLOGD("Invalid movement, distance = [{}, {}], disInch = %f", moveDistance.x, moveDistance.y);`

yes, be careful covering all

**Comment by halx99** [comment]
The fmtlib official docs: https://fmt.dev/latest/index.html may help you

**Comment by aismann** [comment]
@halx99 @rh101 
Please check/review each "change" immedily => So I can correct it faster.

**Comment by aismann** [comment]
@halx99 Why:     =>    build / winuwp (pull_request) Failing after 21m

File: extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp [hunk 1]
```diff
@@ -111,7 +111,7 @@ void PUMaterialCache::addMaterial(PUMaterial* material)
     {
         if (iter->name == material->name)
         {
-            AXLOG("warning: Material has existed (FilePath: %s,  MaterialName: %s)", material->fileName.c_str(),
+            AXLOGD("warning: Material has existed (FilePath: {},  MaterialName: {})", material->fileName.c_str(),
```
Comment (halx99) [inline extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp]: .c_str() can be remove, and other location

File: extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp [hunk 2]
```diff
@@ -187,7 +187,7 @@ bool PUMaterialCache::loadMaterialsFromSearchPaths(std::string_view fileFolder)
 
     if (!(d = opendir(fileFolder.data())))
     {
-        AXLOG("error opendir %s!!!\n", fileFolder.data());
+        AXLOGD("error opendir {}!!!\n", fileFolder.data());
```
Comment (halx99) [inline extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp]: .data() can be remove

File: extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp (hunk 1 above)
Comment (aismann) [reply extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp]: Ah thats the reason why this be not "changed"?
image

File: extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp (hunk 2 above)
Comment (aismann) [reply extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp]: I think its easier make for each kind of %... an own PR

Comment (rh101) [reply extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp]: > I think its easier make for each kind of %... an own PR

If you mean create a new PR per change related to this, then please don't do that. They should all go in this existing PR, and new PRs created will just spam everyone that has notifications turned on for this repo.

Comment (halx99) [reply extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp]: > > I think its easier make for each kind of %... an own PR
> 
> If you mean create a new PR per change related to this, then please don't do that. They should all go in this existing PR, and new PRs created will just spam everyone that has notifications turned on for this repo.

agree, don't create separate PR

Comment (aismann) [reply extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp]: @halx99 
Any hint:
    AXLOGD("Add sprite %0.2f x %02.f", p.x, p.y);
    AXLOGD("Add sprite {} x {}.f", p.x, p.y);  <<<< ?

Comment (aismann) [reply extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp]: I think was forgotten?
image
image
...some more...

Comment (rh101) [reply extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp]: > @halx99 Any hint: AXLOGD("Add sprite %0.2f x %02.f", p.x, p.y); AXLOGD("Add sprite {} x {}.f", p.x, p.y); <<<< ?

`AXLOGD("Add sprite {:.2f} x {:.2f}", p.x, p.y);`

There's a good bit of info here on different format specifiers:
https://hackingcpp.com/cpp/libs/fmt.html

Comment (aismann) [reply extensions/Particle3D/src/Particle3D/PU/PUMaterialManager.cpp]: Another issue:
   `     AXLOGD("downloading... {}%%", percent);`
        
        I think correct is:
`                AXLOGD("downloading... {}%", percent);`

File: extensions/cocostudio/src/cocostudio/SGUIReader.cpp [hunk 3]
```diff
@@ -1470,13 +1470,13 @@ Widget* WidgetPropertiesReader0300::widgetFromBinary(CocoLoader* cocoLoader, stE
             customJsonDict.Parse<0>(customProperty);
             if (customJsonDict.HasParseError())
             {
-                AXLOG("GetParseError %d\n", customJsonDict.GetParseError());
+                AXLOGD("GetParseError {}\n", customJsonDict.GetParseError());
```
Comment (halx99) [inline extensions/cocostudio/src/cocostudio/SGUIReader.cpp]: Change `AXLOGD("GetParseError {}\n", customJsonDict.GetParseError());` to `AXLOGD("GetParseError {}\n", static_cast<int>(customJsonDict.GetParseError()));` will fix winuwp compile errors, and remember change all occurrences.

Comment (aismann) [reply extensions/cocostudio/src/cocostudio/SGUIReader.cpp]: @halx99 
Once again: Any hints?

`build / winuwp (pull_request) Failing after 30m`

**Comment by aismann** [comment]
I belive I have all.

One question:
Should the "`prinf(...)`" also be changed?

```
void RenderCommand::printID()
{
    printf("Command Depth: %f\n", _globalOrder);
}
```
there are some more (e.g tests):
image

**Comment by halx99** [comment]
> I belive I have all.
> 
> One question: Should the "`prinf(...)`" also be changed?
> 
> ```
> void RenderCommand::printID()
> {
>     printf("Command Depth: %f\n", _globalOrder);
> }
> ```
> 
> there are some more (e.g tests): image

yes

File: extensions/scripting/lua-bindings/manual/extension/axlua_extension_manual.cpp [hunk 4]
```diff
@@ -48,7 +48,7 @@ class LuaRefMap : public Object
 
     ~LuaRefMap()
     {
-        AXLOGI("deallocing LuaRefMap: {}", this);
```
Comment (halx99) [inline extensions/scripting/lua-bindings/manual/extension/axlua_extension_manual.cpp]: Should be `AXLOGI("deallocing LuaRefMap: {}", fmt::ptr(this));`

Note: all non fmtlib style LOG macros shoud deprecated

**Comment by aismann** [comment]
@halx99 
Should I replace also .\extensions\spine\runtime ?

image

**Comment by aismann** [comment]
If no more changes want by you: I'm finished. => Its Ready for review now

Not changed (printf(...)):
image
and at the moment 'spine'=> Waiting to your feedback
image

File: extensions/Particle3D/src/Particle3D/PU/PUScriptLexer.h [hunk 5]
```diff
@@ -30,6 +31,7 @@
 #include "base/Object.h"
 #include <vector>
 #include <string>
+#include "axmol.h"
```
Comment (halx99) [inline extensions/Particle3D/src/Particle3D/PU/PUScriptLexer.h]:
```suggestion
#include "base/Logging.h"
```

File: extensions/Particle3D/src/Particle3D/PU/PUScriptParser.h [hunk 6]
```diff
@@ -32,6 +33,7 @@
 #include <list>
 #include <map>
 
+#include "axmol.h"
```
Comment (halx99) [inline extensions/Particle3D/src/Particle3D/PU/PUScriptParser.h]:
```suggestion
#include "base/Logging.h"
```

File: extensions/cocostudio/src/cocostudio/ActionTimeline/BoneNode.cpp [hunk 7]
```diff
@@ -182,7 +182,7 @@ void BoneNode::addToBoneList(BoneNode* bone)
                     _rootSkeleton->_subBonesOrderDirty = true;
                 }
                 else
-                    AXLOG("already has a bone named %s in skeleton %s", bonename.data(),
+                    AXLOGD("already has a bone named {} in skeleton {}", bonename,
                           _rootSkeleton->getName().data());
```
Comment (halx99) [inline extensions/cocostudio/src/cocostudio/ActionTimeline/BoneNode.cpp]:
```suggestion
                          _rootSkeleton->getName());
```

File: extensions/scripting/lua-bindings/manual/ComponentLua.cpp [hunk 8]
```diff
@@ -154,7 +154,7 @@ bool ComponentLua::getLuaFunction(std::string_view functionName)
     int type = lua_type(l, -1);
     //    if (type != LUA_TFUNCTION)
     //    {
-    //        AXLOG("can not get %s function from %s", functionName.c_str(), _scriptFileName.c_str());
+    //        AXLOGD("can not get {} function from {}", functionName.c_str(), _scriptFileName.c_str());
```
Comment (halx99) [inline extensions/scripting/lua-bindings/manual/ComponentLua.cpp]:
```suggestion
    //        AXLOGD("can not get {} function from {}", functionName, _scriptFileName);
```

File: extensions/scripting/lua-bindings/manual/ComponentLua.cpp [hunk 9]
```diff
@@ -194,7 +194,7 @@ bool ComponentLua::loadAndExecuteScript()
     int type = lua_type(l, -1);
     if (type != LUA_TTABLE)
     {
-        AXLOG("%s should return a table, or the script component can not work currectly", _scriptFileName.c_str());
+        AXLOGD("{} should return a table, or the script component can not work currectly", _scriptFileName.c_str());
```
Comment (halx99) [inline extensions/scripting/lua-bindings/manual/ComponentLua.cpp]:
```suggestion
        AXLOGD("{} should return a table, or the script component can not work currectly", _scriptFileName);
```

File: extensions/scripting/lua-bindings/manual/LuaStack.cpp [hunk 10]
```diff
@@ -685,7 +685,7 @@ int LuaStack::luaLoadChunksFromZIP(lua_State* L)
         auto zip = ZipFile::createFromFile(zipFilePath);
         if (zip)
         {
-            AXLOG("lua_loadChunksFromZIP() - load zip file: %s", zipFilePath.c_str());
+            AXLOGD("lua_loadChunksFromZIP() - load zip file: {}", zipFilePath.c_str());
```
Comment (halx99) [inline extensions/scripting/lua-bindings/manual/LuaStack.cpp]:
```suggestion
            AXLOGD("lua_loadChunksFromZIP() - load zip file: {}", zipFilePath);
```

File: extensions/scripting/lua-bindings/manual/LuaStack.cpp [hunk 11]
```diff
@@ -716,7 +716,7 @@ int LuaStack::luaLoadChunksFromZIP(lua_State* L)
                             character = '.';
                         }
                     }
-                    AXLOG("[luaLoadChunksFromZIP] add %s to preload", filename.c_str());
+                    AXLOGD("[luaLoadChunksFromZIP] add {} to preload", filename.c_str());
```
Comment (halx99) [inline extensions/scripting/lua-bindings/manual/LuaStack.cpp]:
```suggestion
                    AXLOGD("[luaLoadChunksFromZIP] add {} to preload", filename);
```

File: extensions/scripting/lua-bindings/manual/LuaStack.cpp [hunk 12]
```diff
@@ -725,15 +725,15 @@ int LuaStack::luaLoadChunksFromZIP(lua_State* L)
...
+            AXLOGD("lua_loadChunksFromZIP() - loaded chunks count: {}", count);
             lua_pop(L, 2);
             lua_pushboolean(L, 1);
 
             delete zip;
         }
         else
         {
-            AXLOG("lua_loadChunksFromZIP() - not found or invalid zip file: %s", zipFilePath.c_str());
+            AXLOGD("lua_loadChunksFromZIP() - not found or invalid zip file: {}", zipFilePath.c_str());
```
Comment (halx99) [inline extensions/scripting/lua-bindings/manual/LuaStack.cpp]:
```suggestion
            AXLOGD("lua_loadChunksFromZIP() - not found or invalid zip file: {}", zipFilePath);
```

File: tests/cpp-tests/Source/UITest/CocoStudioGUITest/UIRichTextTest/UIRichTextTest.cpp [hunk 13]
```diff
@@ -184,10 +184,10 @@ bool UIRichTextTest::init()
...
         std::string str1 = config->getValue("Chinese").asString();
         std::string str2 = config->getValue("Japanese").asString();
-        AXLOG("str1:%s ascii length = %d, utf8 length = %d, substr = %s", str1.c_str(),
+        AXLOGD("str1:{} ascii length = {}, utf8 length = {}, substr = {}", str1,
               static_cast<int32_t>(str1.length()), StringUtils::getCharacterCountInUTF8String(str1),
               Helper::getSubStringOfUTF8String(str1, 0, 5).c_str());
-        AXLOG("str2:%s ascii length = %d, utf8 length = %d, substr = %s", str2.c_str(),
+        AXLOGD("str2:{} ascii length = {}, utf8 length = {}, substr = {}", str2,
               static_cast<int32_t>(str2.length()), StringUtils::getCharacterCountInUTF8String(str2),
               Helper::getSubStringOfUTF8String(str2, 0, 2).c_str());
```
Comment (halx99) [inline tests/cpp-tests/Source/UITest/CocoStudioGUITest/UIRichTextTest/UIRichTextTest.cpp]:
```suggestion
              Helper::getSubStringOfUTF8String(str2, 0, 2));
```

File: tests/cpp-tests/Source/UITest/CocoStudioGUITest/UIRichTextTest/UIRichTextTest.cpp [hunk 14]
```diff
@@ -184,10 +184,10 @@ bool UIRichTextTest::init()
 
         std::string str1 = config->getValue("Chinese").asString();
         std::string str2 = config->getValue("Japanese").asString();
-        AXLOG("str1:%s ascii length = %d, utf8 length = %d, substr = %s", str1.c_str(),
+        AXLOGD("str1:{} ascii length = {}, utf8 length = {}, substr = {}", str1,
               static_cast<int32_t>(str1.length()), StringUtils::getCharacterCountInUTF8String(str1),
               Helper::getSubStringOfUTF8String(str1, 0, 5).c_str());
```
Comment (halx99) [inline tests/cpp-tests/Source/UITest/CocoStudioGUITest/UIRichTextTest/UIRichTextTest.cpp]:
```suggestion
              Helper::getSubStringOfUTF8String(str1, 0, 5));
```

**Review by halx99 — CHANGES_REQUESTED** [review]

**Review by halx99 — APPROVED** [review]

**Comment by aismann** [comment]
@halx99 
I belive some copyright lines be missed.
Should I add it too?

**Comment by halx99** [comment]
> @halx99 I belive some copyright lines be missed. Should I add it too?

sure, you can

**Comment by aismann** [comment]
Add lot of `Copyright (c) 2019-present Axmol Engine contributors (see AUTHORS.md).` lines

@halx99 @rh101 
Im finished. No more changes planed.
Please merge after review.

**Comment by aismann** [comment]
@halx99 
I add the copyright here for mesh.cpp/h too.
Merge of PR: `Add interfaces for easy expansion #2007 ` can start too?

I have no direct access for editng to PR #2007

**Comment by halx99** [comment]
> @halx99 I add the copyright here for mesh.cpp/h too. Merge of PR: `Add interfaces for easy expansion #2007 ` can start too?
> 
> I have no direct access for editng to PR #2007

You don't need care about it

**Pull request merged**